app/anti_virus_check.py

```diff
@@ -0,0 +1,158 @@
+        data_id = self._send_for_anti_virus_check(payload.file_name, payload.decoded_contents)
+        self.bound_logger.info("Sent for A/V check", data_id=data_id)
+
+        # this loop will block the consumer until the anti virus finishes
+        # in a future iteration we should make this asynchronous so that we
+        # can process multiple messages
+        attempts = 0
+        while attempts <= settings.ANTI_VIRUS_MAX_ATTEMPTS:
+            attempts += 1
+            results = self._get_anti_virus_result(data_id)
+            if not results.ready:
+                time.sleep(settings.ANTI_VIRUS_WAIT_TIME)
+            elif not results.safe:
+                error = "Unsafe file detected".format(case_id=payload.case_id, filename=payload.file_name)
+                self._write_scan_report(results, payload.file_name)
+                self.bound_logger.error(error)
+                raise QuarantinableError()
+            else:
+                self.bound_logger.debug(
+                    "File has been virus checked and confirmed safe", case_id=payload.case_id, filename=payload.file_name)
+                return True
+
+        # out of attempts raise retryable error to force the response back to the queue.
+        self.bound_logger.error(
+            "Unable to get results of Anti-virus scan for case id {} and file {} ".format(payload.case_id,
+                                                                                          payload.file_name))
+        raise RetryableError()
+
+    def _add_api_key(self, headers):
+        # the API key is only needed for the online cloud service
+        if settings.ANTI_VIRUS_API_KEY:
+            self.bound_logger.debug("Setting A/V API key")
+            headers['apikey'] = settings.ANTI_VIRUS_API_KEY
+
+    def _check_av_response(self, response):
+        try:
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            self.bound_logger.error("Error received for A/V server", error=e)
+            if response.status_code == 401:
+                self.bound_logger.critical("Invalid OPSWAT API Key - unable to continue")
+                raise RetryableError()
+            elif response.status_code == 403:
+                self.bound_logger.warning("OPSWAT API Rejected request may have hit usage limit - unable to continue")
+                raise RetryableError()
+            elif response.status_code == 404:
+                # this could mean that the primary A/V server has failed and we've failed over to the backup
+                # in this scenario we need to start over
+                self.bound_logger.critical("OPSWAT AV does not know about this scan - the primary server may have failed")
+                raise RetryableError()
+            elif 400 <= response.status_code < 500:
+                raise BadMessageError()
+            elif 300 < response.status_code >= 500:
```

Comment:
`raise_for_status` only raises `HTTPError` for status codes in the 400 -> 599 range - http://docs.python-requests.org/en/master/_modules/requests/models/#Response.raise_for_status